src/yaml-support/yaml-schema.ts

```diff
@@ -0,0 +1,165 @@
+                const apiVersion = rootMapping.mappings.find((map) => map.key && map.key.raw === 'apiVersion');
+                const kind = rootMapping.mappings.find((map) => map.key && map.key.raw === 'kind');
+                if (apiVersion && kind) {
+                    choices.push(apiVersion.value.raw + GROUP_VERSION_KIND_SEPARATOR + kind.value.raw);
+                }
+            });
+            choices = _.uniq(choices);
+            if (choices.length > 1) {
+                return 'kubernetes://combine/' + _.uniq(choices).join('+');
+            }
+            if (choices.length === 1) {
+                return 'kubernetes://manifest/'  + choices[0];
+            }
+            return undefined;
+        }
+    }, (uri) => {
+        let _uri = Uri.parse(uri);
+        if (!_uri.path || ! _uri.path.startsWith('/')) {
+            return '';
+        }
+        if (_uri.path.includes('+')) {
+            const manifestRef = _uri.path.slice(1).split('+').map((r) => {
+                return {$ref: `kubernetes://manifest/${r}`};
+            });
+            return jsonToString({ oneOf: [...manifestRef] });
+        }
+        let manifestType = _uri.path.slice(1);
+        //apiVersionKind[0] is apiVersion and apiVersionKind[1] is kind
+        const apiVersionKind = manifestType.split(GROUP_VERSION_KIND_SEPARATOR);
+        if (_definitions[manifestType]) {
+            return jsonToString(_definitions[manifestType]);
+        }
+        const matchedSchema = _.values(_definitions).find((schema) => schema.apiVersion && schema.kind && apiVersionKind[1] === schema.kind &&
+                apiVersionKind[0] === schema.apiVersion);
+        if (matchedSchema) {
+            return jsonToString(matchedSchema);
+        }
+        return undefined;
+    });
+}
+
+// convert '#/definitions/com.github.openshift.origin.pkg.build.apis.build.v1.ImageLabel' to
+// 'com.github.openshift.origin.pkg.build.apis.build.v1.ImageLabel'
+const getNameInDefinitions = ($ref: string): string => {
+    const prefix = '#/definitions/';
+    if ($ref.includes(prefix)) {
+        return $ref.slice($ref.indexOf(prefix) + prefix.length);
+    } else {
+        throw new Error('bad reference name.');
+    }
+};
+
+// convert $ref = '#/definitions/io.k8s.kubernetes.pkg.apis.extensions.v1beta1.HTTPIngressPath' to
+//'$ref': 'kubernetes://manifest/io.k8s.kubernetes.pkg.apis.extensions.v1beta1.httpingresspath'
+function patchOnRef(node) {
+    if (!node) {
+        return;
+    }
+    if (_.isArray(node)) {
+        for (let subItem of node) {
+            patchOnRef(subItem);
+        }
+    }
+    if (!_.isObject(node)) {
+        return;
+    }
+    for(let key of Object.keys(node)) {
+        patchOnRef(node[key]);
+    }
+
+    const value = node;
+    if (value && value.$ref) {
+        const match = _definitions[getNameInDefinitions(value.$ref.toLowerCase())];
+        if (match) {
+            value.$ref = `kubernetes://manifest/${match.name.toLowerCase()}`;
```

Comment:
There's `toLowerCase`-ing going on here which I didn't see in the other references to `kubernetes://manifest`.  I'm assuming it does all line up somewhere behind the scenes, but I feel a bit at a loss as to how.  I realise that this sort of node traversal code is inherently abstract and hard to communicate clearly to readers, but it would be good if you could refactor it a bit to make the nature of the URIs and transformations clearer - introducing a few helper functions and un-inlining some of the blocks of code would go a long way I think.